extmod/asyncio/stream.py

```diff
@@ -146,14 +175,24 @@ async def _serve(self, s, cb):
             except:
                 # Ignore a failed accept
                 continue
+            if s2:
+                if ssl and _ssl:
+                    if isinstance(ssl, _ssl.SSLContext):
+                        s2.setblocking(True)
+                        try:
+                            s2 = ssl.wrap_socket(s2, server_side=True)
```

Comment:
Can it use `do_handshake_on_connect=False`?  And then make the socket non-blocking.